docker_cli/image.py

```diff
@@ -0,0 +1,430 @@
+                    overload)
+
+from .exceptions import CommandNotFoundOnImageError
+
+
+class Image:
+    """A class which holds a reference to a given docker image and provides
+    an interface by which to interact with that image.
+
+    -   Capable of building docker images from dockerfiles or dockerfile-
+        formatted strings via the build classmethods.
+    -   Capable of acquiring information about the given image (see property
+        methods below).
+    -   Capable of running commands on containers built from the image using
+        the run method.
+    -   All docker build and run methods on the class can recive a file-like
+        object, (io.TextIOBase or some subclass of it,) and they will write the
+        outputs from docker into the file to enable logging.
+
+    NOTE: This class does not work correctly if the image built does not have
+        bash installed.
+    """
+    Self = TypeVar("Self", bound="Image")
+
+    def __init__(self, name_or_id: str):
+        """Initialize this Image. Connects the class with its ID.
+
+        Parameters
+        ----------
+        name_or_id : str
+            A name or ID by which to find this image using docker inspect.
+        """
+        self._id = get_image_id(name_or_id)
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        context: os.PathLike[str],
+        tag: str,
+        *,
+        dockerfile: os.PathLike[str],
+        output_file: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ):
+        """Build a Dockerfile at the given path with the given name, then
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        context : str
+            The path to the file containing the Dockerfile
+        tag : str
+            A name for the image
+        dockerfile : os.PathLike, optional
+            The location of the Dockerfile to build, relative to the position
+            held at the context argument described above.
+        output_file : io.TextIOBase, optional
+            A file-like object that the stderr output of the docker build
+            program will be written to. If None, the method will not output to
+            a file. Defaults to None.
+        network : str
+            The name of the network associated with the docker image to be
+            built. Defaults to "host"
+        no-cache: bool
+            A boolean designating whether or not the docker build should use
+            the cache
+
+        Returns
+        -------
+        Image
+            An Image class instance that references a Docker image built from
+            the Dockerfile at context/dockerfile
+
+        Raises
+        -------
+        CalledProcessError
+            Raised if the docker build command fails. Holds the value returned
+            by the docker build command in its returncode attribute, if not 0.
```

Comment:
>`Holds the value returned by the docker build command in its returncode attribute, if not 0.`

We don't need to describe the contents of the exception here -- we'd be repeating ourselves a lot if we try to include these details everywhere a `CalledProcessError` could be raised. Users can look at the subprocess module documentation if they want more info about the exception.

```suggestion
            If the docker build command fails.
```